src/plugins/footnotes/widgets.ts

```diff
@@ -139,8 +137,36 @@ export const deleteFootnoteWidget =
             })
           }
         }
+        // Apply the transaction to delete the footnote
 
         view.dispatch(tr)
+
+        // Check if all footnotes within `footnotesElement` have been deleted and remove the footnotesElement if true
+
+        if (pos) {
+          const footnotesElement = findParentNodeClosestToPos(
+            view.state.doc.resolve(pos),
+            (node) => node.type === schema.nodes.footnotes_element
+          )
+
+          if (footnotesElement) {
+            let allFootnotesDeleted = true
+
+            // Check if all footnote in footnotesElement are deleted
+            footnotesElement.node.forEach((child) => {
+              if (child.type === schema.nodes.footnote && !isDeleted(child)) {
+                allFootnotesDeleted = false
+              }
+            })
```

Comment:
lets use descendant here, and exit the loop when there is at lease one item not deleted

Reply:
done